### PR TITLE
tests for assets pallet

### PR DESCRIPTION
**Pull request by ferrell-code**
A few tests for assets pallet relates to #8470 :).

### CODE REVIEW

File: frame/assets/src/tests.rs [hunk 1]
```diff
@@ -556,3 +556,76 @@ fn imbalances_should_work() {
...
+fn force_asset_status_should_work(){
+	new_test_ext().execute_with(|| {
+		Balances::make_free_balance_be(&1, 10);
+		assert_ok!(Assets::create(Origin::signed(1), 0, 1, 30));
+		assert_ok!(Assets::mint(Origin::signed(1), 0, 1, 100));
+
+		//force asset status to change min_balance > balance 
+		assert_ok!(Assets::force_asset_status(Origin::root(), 0, 1, 1, 1, 1, 101, true, false));
+		assert_eq!(Assets::balance(0, 1), 100);
+		//asset in practice becomes frozen until balance >= min_balance
```
Comment (ferrell-code) [inline frame/assets/src/tests.rs:610]: Allows for accounts to have less than min balance, not sure if that is problematic or not

File: frame/assets/src/tests.rs [hunk 2]
```diff
@@ -556,3 +556,76 @@ fn imbalances_should_work() {
...
+		);
+
+		//force clear metadata works
+		assert!(Metadata::<Test>::contains_key(0));
+		assert_ok!(Assets::force_clear_metadata(Origin::root(), 0));
+		assert!(!Metadata::<Test>::contains_key(0));
+
+		//Error handles clearing non-existent asset class
+		assert_noop!(Assets::force_clear_metadata(Origin::root(), 1), Error::<Test>::Unknown);
+
```
Comment (bkchr) [inline frame/assets/src/tests.rs:596]:
```suggestion
```

File: frame/assets/src/tests.rs [hunk 3]
```diff
@@ -556,3 +556,76 @@ fn imbalances_should_work() {
...
+		);
+
+		//account drains to completion when funds dip below min_balance
+		assert_ok!(Assets::force_asset_status(Origin::root(), 0, 1, 1, 1, 1, 90, true, false));
+		assert_ok!(Assets::transfer(Origin::signed(1), 0, 2, 50));
+		assert_eq!(Assets::balance(0, 1), 0);
+		assert_eq!(Assets::balance(0, 2), 100);
+		assert_eq!(Assets::total_supply(0), 100);
+	});
+
```
Comment (bkchr) [inline frame/assets/src/tests.rs:629]:
```suggestion
```

File: frame/assets/src/tests.rs (hunk 1 above)
Comment (shawntabrizi) [reply frame/assets/src/tests.rs:609]: What happens when you transfer 1 asset to this user. Will they accept the 1 asset or will it kill the account?

**Comment by shawntabrizi** [comment]
@ferrell-code please add one check about the behavior of an account with below min-balance after `force_asset_status`, then it all looks good to me. :)

**Comment by shawntabrizi** [comment]
I dont think you quite tested what I was interested to see.

A asset has min balance 30, and the user has 50 balance.

Then we increase asset min balance to 100. Currently the user's account is alive with 50 balance.

Then we give the user 1 balance so they have 51. Their new balance, still under the min balance, should trigger cleanup.

**Comment by shawntabrizi** [comment]
But otherwise I am happy to merge. Very nice tests 👍

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
@ferrell-code do you have a polkadot or kusama address? Please post it here.

**Comment by ferrell-code** [comment]
Not with any thing in them :eyes:, use these 15zKd25HoSDPHCn1HjuuHTioXLT7PA2pZQ8gYjFsHW4r96qS
DSSYqgJqdE6HmFAXpiq3y6eZ5FYZ3ZxmX3joRBkh345mxEN

**Review by bkchr — APPROVED** [review]

**Pull request merged**